src/gevent/_hub_primitives.py

```diff
@@ -133,40 +130,49 @@ def __iter__(self):
     def __next__(self):
         self._begin()
 
-        if self._count == 0:
+        if not self._objects or self._count == 0:
```

Comment:
We can't hash the objects and we can't remove them. This needs to go back to the way it was.